nifi-nar-bundles/nifi-scripting-bundle/nifi-scripting-processors/src/main/java/org/apache/nifi/processors/script/ScriptedTransformRecord.java

```diff
@@ -17,7 +17,6 @@
 
```

Comment:
Optional: There are some improvement opportunities in this class beyond your modifications, feel free to pass.
1. Method extraction to reduce cognitive complexity: 158-185
2. Method extraction to reduce cognitive complexity: 187-213
3. Removing redundant array creation from the logging method towards the bottom.
4. To throw more specific exceptions instead of Runtime exceptions.

Reply:
- As for points 1 and 2, the same applies as above.
- For point 3> I am not sure which ones you are referring, but the ones in lines 254, 262 and 270: the content of the array is different in every case thus, we cannot spare them
- `RuntimeException`: in general I agree with you and it could be a good later improvement point but as this ticket ads a bulk of new functionality I would find it unlucky to functionally change already existing behaviour. That easily ends up with regression issues.

Reply:
3. Since _void log(String msg, Object... os)_ methods are added, _void trace(String msg, Object[] os)_ should be deprecated. The object array wrapping has become redundant and shall be removed with no remorse and forever, whenever seen.

Reply:
My IDEA actually is showing it to me: "Redundant array creation for calling varargs method ". Yours isn't?